spec/gem_helper.rb

```diff
@@ -6,13 +6,14 @@
 require './lib/regexp-examples.rb'
 require 'helpers'
 require 'pry'
+require 'warning'
 
 # Several of these tests (intentionally) use "weird" regex patterns,
 # that spam annoying warnings when running.
 # E.g. warning: invalid back reference: /\k/
 # and  warning: character class has ']' without escape: /[]]/
 # This config disables those warnings.
-$VERBOSE = nil
+Warning.ignore(//, __dir__)
```

Comment:
Never heard of [this](https://github.com/jeremyevans/ruby-warning) before, cool!

Maybe some day I'll go through the specs to be more explicit about which warnings are intentionally being ignored.